static/src/javascripts/projects/common/modules/discussion/comment-count.js

```diff
@@ -0,0 +1,141 @@
+// @flow
+import fastdom from 'fastdom';
+import fetchJSON from 'lib/fetch-json';
+import { integerCommas } from 'lib/formatters';
+import mediator from 'lib/mediator';
+import { inlineSvg } from 'common/views/svgs';
+
+type IndexedElements = { string: HTMLElement } | {};
+
+const ATTRIBUTE_NAME: string = 'data-discussion-id';
+const COUNT_URL: string = '/discussion/comment-counts.json?shortUrls=';
+
+const getTemplate = (
+    vals: { url: string, icon: string, count: string },
+    type: string
+): string => {
+    const { url, icon, count } = vals;
+
+    if (type === 'content') {
+        return `<a href="${url}" data-link-name="Comment count" class="commentcount2 tone-colour">
+                    <h3 class="commentcount2__heading">${icon} <span class ="commentcount2__text u-h">Comments</span></h3>
+                    <span class="commentcount2__value tone-colour js_commentcount_actualvalue">${count}</span>
+                </a>`;
+    }
+
+    if (type === 'contentImmersive') {
+        return `<a href="${url}" data-link-name="Comment count" class="commentcount2 tone-colour">
+                    ${icon}<span class="commentcount__value">${count}</span> Comments
+                </a>`;
+    }
+
+    return `<a class="fc-trail__count fc-trail__count--commentcount" href="${url}" data-link-name="Comment count">${icon} ${count}</a>`;
+};
+
+const getElementsIndexedById = (context: HTMLElement): IndexedElements => {
+    const elements = context.querySelectorAll(`[${ATTRIBUTE_NAME}]`);
```

Comment:
That's an expensive selector! oO

Reply:
Yep, the original code used a `qwery` lookup, which would've eventually used `querySelectorAll`.